### ISSUE TITLE
Unexpected server message from rfb.js several hours after establishing connection

### ISSUE THREAD

**Issue by NithyeshS** [novnc/noVNC#813]
I am using noVNC viewer to connect to libVNCServer. I am able to connect to the server and view the screen. Several hours later I get the following error on the browser console

`Failed while connected: Unexpected server message (Unsupported encoding 4257825)`

On the server, the client doesn't get disconnected.

`22/04/2017 01:21:08 Got connection from client 192.168.0.29`
`22/04/2017 01:21:08   other clients:`
`22/04/2017 01:21:08 Got 'ws' WebSockets handshake`
`22/04/2017 01:21:08 Got protocol: binary`
`22/04/2017 01:21:08   - webSocketsHandshake: using binary/raw encoding`
`22/04/2017 01:21:08   - WebSockets client version hybi-13`
`22/04/2017 01:21:08 Client Protocol Version 3.8`
`22/04/2017 01:21:08 Protocol version sent 3.8, using 3.8`
`22/04/2017 01:21:09 rfbProcessClientSecurityType: executing handler for type 1`
`22/04/2017 01:21:09 rfbProcessClientSecurityType: returning securityResult for client rfb version >= 3.8`
`22/04/2017 01:21:10 Pixel format for client 192.168.0.29:`
`22/04/2017 01:21:10   32 bpp, depth 24, little endian`
`22/04/2017 01:21:10   true colour: max r 255 g 255 b 255, shift r 16 g 8 b 0`
`22/04/2017 01:21:11 rfbProcessClientNormalMessage: ignoring unsupported encoding type tight`
`22/04/2017 01:21:11 rfbProcessClientNormalMessage: ignoring unsupported encoding type tightPng`
`22/04/2017 01:21:11 rfbProcessClientNormalMessage: ignoring unsupported encoding type QualityLevel6`
`22/04/2017 01:21:11 rfbProcessClientNormalMessage: ignoring unsupported encoding type CompressLevel9`
`22/04/2017 01:21:11 Enabling NewFBSize protocol extension for client 192.168.0.29`
`22/04/2017 01:21:11 Enabling LastRect protocol extension for client 192.168.0.29`
`22/04/2017 01:21:11 Enabling full-color cursor updates for client 192.168.0.29`
`22/04/2017 01:21:11 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFEFE)`
`22/04/2017 01:21:11 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFECC)`
`22/04/2017 01:21:11 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFEC8)`
`22/04/2017 01:21:11 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFEC7)`
`22/04/2017 01:21:11 Using hextile encoding for client 192.168.0.29`

I was hoping someone could help me debug the issue and solve it

Thanks.

**Comment by CendioOssman** [comment]
This is known, but I noticed we didn't already have an issue for it. Unfortunately all attempts at debugging it has failed. What's missing is a reliable way to reproduce it.

**Comment by DirectXMan12** [comment]
if you can reliably reproduce, a recording via websockify would be tremendously useful, as it provides a way to replicate the exact sequence of bytes that lead to the issue.

**Comment by samhed** [comment]
Yeah, I've seen this issue quite a number of times through the years, but never found a way to reproduce it

**Comment by CendioOssman** [comment]
I'm afraid recordings don't reproduce this reliably. So it doesn't seem to be a protocol error, but rather some kind of race in our parsing.

**Comment by DirectXMan12** [comment]
So, if I had to take a wild shot in the dark, I'd guess that what's going on is that something returned without consuming the rest of the bytes in the message, or didn't rewind appropriately.  I've fixed a couple of bugs related to that over the years, but it's hard to track down if it doesn't reproduce reliably (although, if we slow down our playback framework dramatically, it should reproduce fairly reliably from a recording, if I'm correct).
